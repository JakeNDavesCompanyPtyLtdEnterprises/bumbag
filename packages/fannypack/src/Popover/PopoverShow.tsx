import * as React from 'react';
import * as PropTypes from 'prop-types';
import { PopoverShowProps as ReakitPopoverShowProps } from 'reakit/ts';

import { PopoverShow as _PopoverShow } from './styled';

export type LocalPopoverShowProps = {
  children: React.ReactNode;
  onClick?(): void;
  show(): void;
};
export type PopoverShowProps = ReakitPopoverShowProps & LocalPopoverShowProps;

export const PopoverShow: React.FunctionComponent<LocalPopoverShowProps> = ({ children, ...props }) => (
  <_PopoverShow {...props}>{children}</_PopoverShow>
);

PopoverShow.propTypes = {
  children: PropTypes.node.isRequired,
  onClick: PropTypes.func,
  show: PropTypes.func.isRequired
};
PopoverShow.defaultProps = {
  onClick: undefined
};

const C: React.FunctionComponent<PopoverShowProps> = PopoverShow;
export default C;
