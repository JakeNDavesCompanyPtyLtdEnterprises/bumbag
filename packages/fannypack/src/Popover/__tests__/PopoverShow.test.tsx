import * as React from 'react';
import render from '../../_utils/tests/render';
import PopoverShow from '../PopoverShow';

it('renders correctly', () => {
  const { container } = render(<PopoverShow show={jest.fn()}>Show</PopoverShow>);
  expect(container.firstChild).toMatchSnapshot();
});
