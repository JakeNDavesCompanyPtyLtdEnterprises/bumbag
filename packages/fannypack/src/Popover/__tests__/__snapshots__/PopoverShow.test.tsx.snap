// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c0:focus:not(:focus-visible) {
  outline: none;
}

<button
  aria-haspopup="true"
  class="c0"
>
  Show
</button>
`;
